### ISSUE TITLE
Saving theme unsuccessful with error

### ISSUE THREAD

**Issue by hyperfekt** [OpenUserCSS/openusercss.org#84]
| Expected behaviour | Actual behaviour |
| ------------------ | ---------------- |
| Saves the theme successfully | Shows only the error `Expected a value of type ``undefined`` for ``__typename`` but received ``"Option"``.`|

<sub>Markdown has no proper backtick escaping D':</sub>

## Steps to Reproduce the Problem
  1. Open this theme for editing: https://openusercss.org/theme/edit/5abb16b770bc4a0b00903b96
  1. Hit 'Save theme'

## Environment details
  - Site version: 0.9.7
  - Browser and version: Firefox 59.0.1, Chromium 63.0.3239.108
  - Installed extensions: None
  - Operating system: Qubes OS (Fedora 26)
----

<details>
  <summary><strong>Stack trace</strong> (click to expand)</summary>

  ```
browser.js:1 [Vue warn]: Do not use built-in or reserved HTML elements as component id: button
console.(anonymous function) @ browser.js:1
browser.js:1 StructError: Expected a value of type `undefined` for `__typename` but received `"Option"`.
    at Function.Struct.assert.value [as assert] (https://openusercss.org/browser.js:1:2181773)
    at Object.Struct [as option] (https://openusercss.org/browser.js:1:2181342)
    at https://openusercss.org/browser.js:1:2976486
    at Array.forEach (<anonymous>)
    at https://openusercss.org/browser.js:1:2976450
    at tryCatch (https://openusercss.org/browser.js:1:2143865)
    at Generator._invoke (https://openusercss.org/browser.js:1:2143470)
    at Generator.prototype.(anonymous function) [as next] (https://openusercss.org/browser.js:1:2144149)
    at step (https://openusercss.org/browser.js:1:2975768)
    at https://openusercss.org/browser.js:1:2975971
    at new Promise (<anonymous>)
    at Store.<anonymous> (https://openusercss.org/browser.js:1:2975687)
    at Store.<anonymous> (https://openusercss.org/browser.js:1:2977778)
    at Array.<anonymous> (https://openusercss.org/browser.js:1:2889996)
    at Store.dispatch (https://openusercss.org/browser.js:1:2898453)
    at Store.dispatch (https://openusercss.org/browser.js:1:2896539)
    at https://openusercss.org/browser.js:1:3057329
    at Generator.next (<anonymous>)
    at step (https://openusercss.org/browser.js:1:3053255)
    at https://openusercss.org/browser.js:1:3053382
    at <anonymous>
  ```
</details>

**Comment by DecentM** [comment]
Thank you for the immaculate use of the issue template!  
The option handling code in 0.9.6 is very shoddy, so I can immediately confirm this.

**Comment by DecentM** [comment]
From #70:

<details>
  <summary><strong>Stack trace 1</strong> (click to expand)</summary>

  ```
browser.js:1 Uncaught (in promise) TypeError: Cannot destructure property `props` of 'undefined' or 'null'.
    at browser.js:1
    at Generator.throw (<anonymous>)
    at step (browser.js:1)
    at browser.js:1
    at <anonymous>
  ```
</details>

<details>
  <summary><strong>Stack trace 2</strong> (click to expand)</summary>

  ```
browser.js:1 TypeError: Cannot read property '2' of null
    at https://openusercss.org/browser.js:1:3141601
    at Array.forEach (<anonymous>)
    at https://openusercss.org/browser.js:1:3141414
    at tryCatch (https://openusercss.org/browser.js:1:2138408)
    at Generator._invoke (https://openusercss.org/browser.js:1:2138013)
    at Generator.prototype.(anonymous function) [as next] (https://openusercss.org/browser.js:1:2138692)
    at step (https://openusercss.org/browser.js:1:3140009)
    at https://openusercss.org/browser.js:1:3140212
    at new Promise (<anonymous>)
    at https://openusercss.org/browser.js:1:3139928
    at https://openusercss.org/browser.js:1:3142168
    at https://openusercss.org/browser.js:1:3050984
    at Generator.next (<anonymous>)
    at step (https://openusercss.org/browser.js:1:3046144)
    at https://openusercss.org/browser.js:1:3046347
    at new Promise (<anonymous>)
    at https://openusercss.org/browser.js:1:3046063
    at Sub.parseUserCSS (https://openusercss.org/browser.js:1:3051349)
    at boundFn (https://openusercss.org/browser.js:1:2748778)
    at Sub.invoker (https://openusercss.org/browser.js:1:2761396)
    at Sub.Vue.$emit (https://openusercss.org/browser.js:1:2829439)
    at Array.editor.on (https://openusercss.org/browser.js:1:3003518)
    at Editor.EventEmitter._signal (https://openusercss.org/browser.js:1:247951)
    at Editor.onDocumentChange (https://openusercss.org/browser.js:1:471960)
    at EditSession.EventEmitter._signal (https://openusercss.org/browser.js:1:247951)
    at EditSession.onChange (https://openusercss.org/browser.js:1:396587)
    at Document.EventEmitter._signal (https://openusercss.org/browser.js:1:247951)
    at Document.applyDelta (https://openusercss.org/browser.js:1:360097)
    at Document.insertMergedLines (https://openusercss.org/browser.js:1:357810)
    at Document.insert (https://openusercss.org/browser.js:1:356288)
    at EditSession.insert (https://openusercss.org/browser.js:1:406944)
    at Editor.insert (https://openusercss.org/browser.js:1:477435)
    at Editor.$handlePaste (https://openusercss.org/browser.js:1:475918)
    at Object.exec (https://openusercss.org/browser.js:1:451092)
    at https://openusercss.org/browser.js:1:439382
    at CommandManager.EventEmitter._emit.EventEmitter._dispatchEvent (https://openusercss.org/browser.js:1:247736)
    at CommandManager.exec (https://openusercss.org/browser.js:1:439925)
    at Editor.onPaste (https://openusercss.org/browser.js:1:475732)
    at HTMLTextAreaElement.onPaste (https://openusercss.org/browser.js:1:219715)
    at HTMLTextAreaElement.wrapped (https://openusercss.org/browser.js:1:2091242)
console.(anonymous function) @ browser.js:1
  ```
</details>

**Comment by DecentM** [comment]
Link #86

**Comment by DecentM** [comment]
From #89 

<details>
  <summary><strong>Stack trace</strong> (click to expand)</summary>

```
  Error: GraphQL error: Cannot read property '2' of null
  Stack trace:
  ApolloError@https://openusercss.org/browser.js:1:78919
  [4]QueryManager QueryManager.prototype.mutate @https://openusercss.org/browser.js:1:102387
  run@https://openusercss.org/browser.js:1:1126531notify @https://openusercss.org/browser.js:1:1126763
  flush@https://openusercss.org/browser.js:1:1058384
```
</details>

**Comment by DecentM** [comment]
This should be fixed with 1.0.0. I'll close this issue, but if it still persists for you, please let me know!
